homeassistant/components/sonos/speaker.py

```diff
@@ -470,7 +475,20 @@ async def async_unseen(self, now: datetime.datetime | None = None) -> None:
             await subscription.unsubscribe()
 
         self._subscriptions = []
-        self.hass.data[DATA_SONOS].ssdp_known.remove(self.soco.uid)
+
+        if not will_reconnect:
+            self.hass.data[DATA_SONOS].ssdp_known.remove(self.soco.uid)
+            self.async_write_entity_states()
+
+    async def async_rebooted(self, soco: SoCo) -> None:
+        """Handle a detected speaker reboot."""
+        _LOGGER.warning(
```

Comment:
```suggestion
        _LOGGER.debug(
```

Reply:
An event like this _should_ be exceptional. However there are some users that seem to be hitting it more often than they should. Visible warnings this has occurred may help them uncover a latent issue in their particular setup.

That being said either way is fine. Do you have strong feelings one way or the other?

Reply:
Let's keep it for now and downgrade if it's too frequent for some users for uncontrollable reasons.